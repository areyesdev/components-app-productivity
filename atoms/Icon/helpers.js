const ICONS_ASSETS_PATH = '/icons',
  iconMap = {
    arrowRight: 'arrow-right',
    angleDown: 'angle-down',
    iconPlus: 'icon-plus',
  },
  iconSize = {
    sm: 15,
    md: 26,
    lg: 40,
    xl: 40,
  }

export const mapType = (type) => `${ICONS_ASSETS_PATH}/${iconMap[type]}.svg`
export const mapSize = (size) => iconSize[size]
