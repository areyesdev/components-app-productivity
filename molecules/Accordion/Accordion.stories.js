import { Accordion, styles } from '.'

import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(Accordion, styles)

export default {
  title: 'Molecules/Accordion',
  component: Accordion,
  args: {
    title: 'Em Sméagol body.',
    children:
      "Fells defeated veins this. There and Back Again. A Hobbit's Tale. Loved lucky Istari difficult lay? Dale fellow's rioting. Blanket Misty Mountain started rival wriggles flatten convinced succumbed closing? Slug town fond pipe-weed flatten recognizes Goblins! Innards pouf admirable freely nothingness sleepless exactly afternoon cruelty Bofur? Craftsmen tested fair Alfrid gets figure words winter's field offer healed squash? These galumphing Grima tread mustn't seen melt ideal picked recover employment. Nazgûl alas luckiest?",
  },
}

export const Default = Template.bind({})

export const AddonPrepend = Template.bind({})
AddonPrepend.args = {
  addons: {
    prepend: 'Dare Erebor forgive most.',
  },
}

export const AddonAppend = Template.bind({})
AddonAppend.args = {
  addons: {
    append: 'But ill-tempered Thengel.',
  },
}
