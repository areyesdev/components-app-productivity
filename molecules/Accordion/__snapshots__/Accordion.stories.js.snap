// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Accordion Addon Append 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="accordion"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="icon color-inverted size-sm background-muted"
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      >
        <svg
          height={25}
          viewBox="0 0 15 9"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="spacer horizontal"
        style={
          Object {
            "height": 15,
          }
        }
      />
      <div
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </div>
      But ill-tempered Thengel.
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/Accordion Addon Prepend 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="accordion"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="icon color-inverted size-sm background-muted"
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      >
        <svg
          height={25}
          viewBox="0 0 15 9"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="spacer horizontal"
        style={
          Object {
            "height": 15,
          }
        }
      />
      Dare Erebor forgive most.
      <div
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </div>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;

exports[`Storyshots Molecules/Accordion Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="accordion"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="icon color-inverted size-sm background-muted"
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      >
        <svg
          height={25}
          viewBox="0 0 15 9"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="spacer horizontal"
        style={
          Object {
            "height": 15,
          }
        }
      />
      <div
        className="paragraph color-base size-md weight-medium"
      >
        Em Sméagol body.
      </div>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 5,
        }
      }
    />
    <div
      className="divider"
    />
  </div>
</div>
`;
